// flex-plugins-utils-env/dist/lib/env.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.isNode = void 0;
var lodash_get_1 = __importDefault(require("lodash.get"));
exports.isNode = function () { return typeof process === 'object' && "" + process === '[object process]'; };
var isWin32 = function () { return exports.isNode() && process.platform === 'win32'; };
var isCI = function () { return exports.isNode() && process.env.CI === 'true'; };
var persistTerminal = function () {
    if (exports.isNode()) {
        process.env.PERSIST_TERMINAL = 'true';
    }
};
var isTerminalPersisted = function () { return exports.isNode() && process.env.PERSIST_TERMINAL === 'true'; };
var isQuiet = function () { return exports.isNode() && process.env.QUIET === 'true'; };
var setQuiet = function () {
    process.env.QUIET = 'true';
};
var isTrace = function () {
    if (exports.isNode()) {
        return process.env.TRACE === 'true';
    }
    if (window.Twilio) {
        return window.Twilio.Flex.Manager.getInstance().configuration.logLevel === 'trace';
    }
    return false;
};
var isDebug = function () {
    if (isTrace()) {
        return true;
    }
    if (exports.isNode()) {
        return process.env.DEBUG === 'true';
    }
    if (window.Twilio) {
        return window.Twilio.Flex.Manager.getInstance().configuration.logLevel === 'debug';
    }
    return false;
};
var setRealm = function (realm) {
    process.env.REALM = realm;
};
var getRealm = function () {
    if (exports.isNode()) {
        return process.env.REALM;
    }
    if (window.Twilio) {
        var region = lodash_get_1.default(window.Twilio.Flex.Manager.getInstance(), 'configuration.sdkOptions.chat.region');
        if (region && region.indexOf('stage') !== -1) {
            return 'stage';
        }
        if (region && region.indexOf('dev') !== -1) {
            return 'dev';
        }
        return '';
    }
    var href = window.location.href;
    if (href && href.indexOf('flex.stage.twilio') !== -1) {
        return 'stage';
    }
    if (href && href.indexOf('flex.dev.twilio') !== -1) {
        return 'dev';
    }
    return '';
};
exports.default = {
    isWin32: isWin32,
    persistTerminal: persistTerminal,
    isTerminalPersisted: isTerminalPersisted,
    isQuiet: isQuiet,
    setQuiet: setQuiet,
    isDebug: isDebug,
    isTrace: isTrace,
    getRealm: getRealm,
    setRealm: setRealm,
    isCI: isCI,
};
