// flex-plugins-utils-logger/dist/lib/boxen.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.success = exports.error = exports.info = exports.warning = exports.print = void 0;
var boxen_1 = __importDefault(require("boxen"));
var log_symbols_1 = __importDefault(require("log-symbols"));
var logger_1 = __importDefault(require("./logger"));
exports.default = boxen_1.default;
exports.print = function (level, msg) {
    var boxed = boxen_1.default(msg, {
        padding: 1,
        margin: 1,
    });
    logger_1.default[level](boxed);
};
exports.warning = function (msg, showSymbol) {
    if (showSymbol === void 0) { showSymbol = true; }
    var sym = log_symbols_1.default.warning;
    if (showSymbol) {
        msg = sym + " " + msg + " " + sym;
    }
    exports.print('warning', msg);
};
exports.info = function (msg, showSymbol) {
    if (showSymbol === void 0) { showSymbol = true; }
    var sym = log_symbols_1.default.info;
    if (showSymbol) {
        msg = sym + " " + msg + " " + sym;
    }
    exports.print('info', msg);
};
exports.error = function (msg, showSymbol) {
    if (showSymbol === void 0) { showSymbol = true; }
    var sym = log_symbols_1.default.error;
    if (showSymbol) {
        msg = sym + " " + msg + " " + sym;
    }
    exports.print('error', msg);
};
exports.success = function (msg, showSymbol) {
    if (showSymbol === void 0) { showSymbol = true; }
    var sym = log_symbols_1.default.success;
    if (showSymbol) {
        msg = sym + " " + msg + " " + sym;
    }
    exports.print('success', msg);
};
