// flex-plugins-utils-logger/dist/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.boxen = void 0;
var _boxen = __importStar(require("./lib/boxen"));
var boxen = {
    error: _boxen.error,
    warning: _boxen.warning,
    info: _boxen.info,
    print: _boxen.print,
};
exports.boxen = boxen;
var logger_1 = require("./lib/logger");
Object.defineProperty(exports, "Logger", { enumerable: true, get: function () { return logger_1.Logger; } });
Object.defineProperty(exports, "coloredStrings", { enumerable: true, get: function () { return logger_1.coloredStrings; } });
var logger_2 = require("./lib/logger");
Object.defineProperty(exports, "logger", { enumerable: true, get: function () { return logger_2.default; } });
var strings_1 = require("./lib/strings");
Object.defineProperty(exports, "strings", { enumerable: true, get: function () { return strings_1.default; } });
Object.defineProperty(exports, "multilineString", { enumerable: true, get: function () { return strings_1.multilineString; } });
Object.defineProperty(exports, "singleLineString", { enumerable: true, get: function () { return strings_1.singleLineString; } });
var table_1 = require("./lib/table");
Object.defineProperty(exports, "table", { enumerable: true, get: function () { return table_1.default; } });
Object.defineProperty(exports, "printArray", { enumerable: true, get: function () { return table_1.printArray; } });
Object.defineProperty(exports, "printObjectArray", { enumerable: true, get: function () { return table_1.printObjectArray; } });
var progress_1 = require("./lib/progress");
Object.defineProperty(exports, "progress", { enumerable: true, get: function () { return progress_1.default; } });
var inquirer_1 = require("./lib/inquirer");
Object.defineProperty(exports, "inquirer", { enumerable: true, get: function () { return inquirer_1.default; } });
Object.defineProperty(exports, "confirm", { enumerable: true, get: function () { return inquirer_1.confirm; } });
Object.defineProperty(exports, "prompt", { enumerable: true, get: function () { return inquirer_1.prompt; } });
Object.defineProperty(exports, "choose", { enumerable: true, get: function () { return inquirer_1.choose; } });
var columnify_1 = require("./lib/columnify");
Object.defineProperty(exports, "columnify", { enumerable: true, get: function () { return columnify_1.default; } });
