// flex-dev-utils/dist/fs.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
var __generator = (this && this.__generator) || function (thisArg, body) {
    var _ = { label: 0, sent: function() { if (t[0] & 1) throw t[1]; return t[1]; }, trys: [], ops: [] }, f, y, t, g;
    return g = { next: verb(0), "throw": verb(1), "return": verb(2) }, typeof Symbol === "function" && (g[Symbol.iterator] = function() { return this; }), g;
    function verb(n) { return function (v) { return step([n, v]); }; }
    function step(op) {
        if (f) throw new TypeError("Generator is already executing.");
        while (_) try {
            if (f = 1, y && (t = op[0] & 2 ? y["return"] : op[0] ? y["throw"] || ((t = y["return"]) && t.call(y), 0) : y.next) && !(t = t.call(y, op[1])).done) return t;
            if (y = 0, t) op = [op[0] & 2, t.value];
            switch (op[0]) {
                case 0: case 1: t = op; break;
                case 4: _.label++; return { value: op[1], done: false };
                case 5: _.label++; y = op[1]; op = [0]; continue;
                case 7: op = _.ops.pop(); _.trys.pop(); continue;
                default:
                    if (!(t = _.trys, t = t.length > 0 && t[t.length - 1]) && (op[0] === 6 || op[0] === 2)) { _ = 0; continue; }
                    if (op[0] === 3 && (!t || (op[1] > t[0] && op[1] < t[3]))) { _.label = op[1]; break; }
                    if (op[0] === 6 && _.label < t[1]) { _.label = t[1]; t = op; break; }
                    if (t && _.label < t[2]) { _.label = t[2]; _.ops.push(op); break; }
                    if (t[2]) _.ops.pop();
                    _.trys.pop(); continue;
            }
            op = body.call(thisArg, _);
        } catch (e) { op = [6, e]; y = 0; } finally { f = t = 0; }
        if (op[0] & 5) throw op[1]; return { value: op[0] ? op[1] : void 0, done: true };
    }
};
var __read = (this && this.__read) || function (o, n) {
    var m = typeof Symbol === "function" && o[Symbol.iterator];
    if (!m) return o;
    var i = m.call(o), r, ar = [], e;
    try {
        while ((n === void 0 || n-- > 0) && !(r = i.next()).done) ar.push(r.value);
    }
    catch (error) { e = { error: error }; }
    finally {
        try {
            if (r && !r.done && (m = i["return"])) m.call(i);
        }
        finally { if (e) throw e.error; }
    }
    return ar;
};
var __spread = (this && this.__spread) || function () {
    for (var ar = [], i = 0; i < arguments.length; i++) ar = ar.concat(__read(arguments[i]));
    return ar;
};
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.getPackageVersion = exports.getDependencyVersion = exports.getPaths = exports._getFlexPluginWebpackPath = exports._getFlexPluginScripts = exports.resolveModulePath = exports.addCWDNodeModule = exports.checkPluginConfigurationExists = exports.findGlobs = exports.findGlobsIn = exports.resolveCwd = exports.rmRfSync = exports.copyTemplateDir = exports.mkdirpSync = exports.readFileSync = exports.findUp = exports.updateAppVersion = exports.readAppPackageJson = exports.getPackageJsonPath = exports.checkFilesExist = exports.writeJSONFile = exports.readPluginsJson = exports.getCliPaths = exports.readJsonFile = exports.getCoreCwd = exports.setCoreCwd = exports.getCwd = exports.setCwd = exports.resolveRelative = exports.readPackageJson = exports._setRequirePaths = exports._require = void 0;
var fs_1 = __importDefault(require("fs"));
var path = __importStar(require("path"));
var os_1 = __importStar(require("os"));
var util_1 = require("util");
var globby_1 = __importDefault(require("globby"));
var mkdirp_1 = __importDefault(require("mkdirp"));
var rimraf_1 = __importDefault(require("rimraf"));
var app_module_path_1 = __importDefault(require("app-module-path"));
var inquirer_1 = require("./inquirer");
exports.default = fs_1.default;
var packageJsonStr = 'package.json';
/**
 * This is an alias for require. Useful for mocking out in tests
 * @param filePath  the file to require
 * @private
 */
/* istanbul ignore next */
// eslint-disable-next-line global-require, @typescript-eslint/no-require-imports, @typescript-eslint/explicit-module-boundary-types
exports._require = function (filePath) { return require(filePath); };
// Set working directory
exports._setRequirePaths = function (requirePath) {
    app_module_path_1.default.addPath(requirePath);
    // Now try to specifically set the node_modules path
    var requirePaths = (require.main && require.main.paths) || [];
    if (!requirePaths.includes(requirePath)) {
        requirePaths.push(requirePath);
    }
};
/**
 * Reads a JSON file
 *
 * @param filePath   the file path to read
 */
exports.readPackageJson = function (filePath) {
    return JSON.parse(fs_1.default.readFileSync(filePath, 'utf8'));
};
/**
 * Builds path relative to the given dir
 * @param dir   the dir
 * @param paths the paths
 */
exports.resolveRelative = function (dir) {
    var paths = [];
    for (var _i = 1; _i < arguments.length; _i++) {
        paths[_i - 1] = arguments[_i];
    }
    if (paths.length === 0) {
        return dir;
    }
    var lastElement = paths[paths.length - 1];
    // Check if last element is an extension
    if (lastElement.charAt(0) !== '.') {
        return path.join.apply(path, __spread([dir], paths));
    }
    // Only one entry as extension
    if (paths.length === 1) {
        return path.join("" + dir + lastElement);
    }
    var secondLastElement = paths[paths.length - 2];
    var remainder = paths.slice(0, paths.length - 2);
    return path.join.apply(path, __spread([dir], __spread(remainder, ["" + secondLastElement + lastElement])));
};
// Working directory
var internalCwd = fs_1.default.realpathSync(process.cwd());
var internalCoreCwd = fs_1.default.realpathSync(process.cwd());
/**
 * Sets the working directory
 * @param p the path to set
 */
exports.setCwd = function (p) {
    internalCwd = p;
    exports._setRequirePaths(path.join(internalCwd, 'node_modules'));
};
/**
 * Returns the working directory
 */
exports.getCwd = function () { return internalCwd; };
/**
 * Sets the core working directory
 * @param p the path to set
 */
exports.setCoreCwd = function (p) {
    internalCoreCwd = p;
    exports._setRequirePaths(path.join(internalCoreCwd, 'node_modules'));
};
/**
 * The core cwd is the working directory of core packages such as flex-plugin-scripts and flex-plugin
 */
exports.getCoreCwd = function () { return internalCoreCwd; };
/**
 * Reads a JSON file (Templated)
 *
 * @param filePath  the file path to read
 */
exports.readJsonFile = function (filePath) {
    return JSON.parse(fs_1.default.readFileSync(filePath, 'utf8'));
};
/**
 * Gets the CLI paths. This is separated out from getPaths because create-flex-plugin also needs to read it,
 * but that script will not have flex-plugin-scripts installed which would cause an exception to be thrown.
 */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
exports.getCliPaths = function () {
    var coreCwd = exports.getCoreCwd();
    var coreNodeModulesDir = exports.resolveRelative(coreCwd, 'node_modules');
    var homeDir = os_1.homedir();
    var cliDir = exports.resolveRelative(homeDir, '/.twilio-cli');
    var flexDir = exports.resolveRelative(cliDir, 'flex');
    return {
        dir: cliDir,
        nodeModulesDir: coreNodeModulesDir,
        flexDir: flexDir,
        pluginsJsonPath: exports.resolveRelative(flexDir, 'plugins.json'),
    };
};
// Read plugins.json from Twilio CLI
exports.readPluginsJson = function () {
    return exports.readJsonFile(exports.getCliPaths().pluginsJsonPath);
};
/**
 * Writes an object as a JSON string to the file
 * @param pth the path to write to
 * @param obj the object to write
 */
// eslint-disable-next-line @typescript-eslint/ban-types
exports.writeJSONFile = function (pth, obj) { return fs_1.default.writeFileSync(pth, JSON.stringify(obj, null, 2)); };
// The OS root directory
var rootDir = os_1.default.platform() === 'win32' ? exports.getCwd().split(path.sep)[0] : '/';
/*
 * Promise version of {@link copyTempDir}
 */
var promiseCopyTempDir = util_1.promisify(exports._require('copy-template-dir'));
/**
 * Checks the provided array of files exist
 *
 * @param files the files to check that they exist
 */
exports.checkFilesExist = function () {
    var files = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        files[_i] = arguments[_i];
    }
    return files.map(fs_1.default.existsSync).every(function (resp) { return resp; });
};
/**
 * Gets package.json path
 */
exports.getPackageJsonPath = function () { return path.join(exports.getCwd(), packageJsonStr); };
/**
 * Reads app package.json from the rootDir.
 */
exports.readAppPackageJson = function () {
    return exports.readPackageJson(exports.getPackageJsonPath());
};
/**
 * Updates the package.json version field
 *
 * @param version the new version
 */
exports.updateAppVersion = function (version) {
    var packageJson = exports.readAppPackageJson();
    packageJson.version = version;
    fs_1.default.writeFileSync(exports.getPackageJsonPath(), JSON.stringify(packageJson, null, 2));
};
/**
 * Finds the closest up file relative to dir
 *
 * @param dir   the directory
 * @param file  the file to look for
 */
exports.findUp = function (dir, file) {
    var resolved = path.resolve(dir);
    if (resolved === rootDir) {
        throw new Error("Reached OS root directory without finding " + file);
    }
    var filePath = path.join(resolved, file);
    if (fs_1.default.existsSync(filePath)) {
        return filePath;
    }
    return exports.findUp(path.resolve(resolved, '..'), file);
};
/**
 * Reads the file
 * @param filePath  the file path
 */
exports.readFileSync = function (filePath) { return fs_1.default.readFileSync(filePath, 'utf8'); };
/**
 * mkdir -p wrapper
 */
exports.mkdirpSync = mkdirp_1.default.sync;
/**
 * Copies a template by applying the variables
 *
 * @param source    the source
 * @param target    the target
 * @param variables the variables
 */
/* istanbul ignore next */
// eslint-disable-next-line @typescript-eslint/ban-types
exports.copyTemplateDir = function (source, target, variables) { return __awaiter(void 0, void 0, void 0, function () {
    return __generator(this, function (_a) {
        return [2 /*return*/, promiseCopyTempDir(source, target, variables)];
    });
}); };
/**
 * rm -rf sync script
 */
exports.rmRfSync = rimraf_1.default.sync;
/**
 * Builds path relative to cwd
 * @param paths  the paths
 */
exports.resolveCwd = function () {
    var paths = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        paths[_i] = arguments[_i];
    }
    return exports.resolveRelative.apply(void 0, __spread([exports.getCwd()], paths));
};
/**
 * Finds globs in any cwd directory
 * @param dir     the cwd to check for patterns
 * @param patterns the patterns
 */
exports.findGlobsIn = function (dir) {
    var patterns = [];
    for (var _i = 1; _i < arguments.length; _i++) {
        patterns[_i - 1] = arguments[_i];
    }
    return globby_1.default.sync(patterns, { cwd: dir });
};
/**
 * Finds globs in the src directory
 * @param patterns the patterns
 */
exports.findGlobs = function () {
    var patterns = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        patterns[_i] = arguments[_i];
    }
    return exports.findGlobsIn.apply(void 0, __spread([path.join(exports.getCwd(), 'src')], patterns));
};
/**
 * Touch ~/.twilio-cli/flex/plugins.json if it does not exist
 * Check if this plugin is in this config file. If not, add it.
 * @param name  the plugin name
 * @param dir   the plugin directory
 * @param promptForOverwrite  whether to prompt for overwrite
 * @return whether the plugin-directory was overwritten
 */
exports.checkPluginConfigurationExists = function (name, dir, promptForOverwrite) {
    if (promptForOverwrite === void 0) { promptForOverwrite = false; }
    return __awaiter(void 0, void 0, void 0, function () {
        var cliPaths, config, plugin, answer, _a;
        return __generator(this, function (_b) {
            switch (_b.label) {
                case 0:
                    cliPaths = exports.getCliPaths();
                    if (!exports.checkFilesExist(cliPaths.pluginsJsonPath)) {
                        exports.mkdirpSync(cliPaths.flexDir);
                        exports.writeJSONFile(cliPaths.pluginsJsonPath, { plugins: [] });
                    }
                    config = exports.readPluginsJson();
                    plugin = config.plugins.find(function (p) { return p.name === name; });
                    if (!plugin) {
                        config.plugins.push({ name: name, dir: dir, port: 0 });
                        exports.writeJSONFile(cliPaths.pluginsJsonPath, config);
                        return [2 /*return*/, true];
                    }
                    if (plugin.dir === dir) {
                        return [2 /*return*/, false];
                    }
                    if (!promptForOverwrite) return [3 /*break*/, 2];
                    return [4 /*yield*/, inquirer_1.confirm("You already have a plugin called " + plugin.name + " located at " + plugin.dir + ". Do you want to update it to " + dir + "?", 'N')];
                case 1:
                    _a = _b.sent();
                    return [3 /*break*/, 3];
                case 2:
                    _a = true;
                    _b.label = 3;
                case 3:
                    answer = _a;
                    if (answer) {
                        plugin.dir = dir;
                        exports.writeJSONFile(cliPaths.pluginsJsonPath, config);
                        return [2 /*return*/, true];
                    }
                    return [2 /*return*/, false];
            }
        });
    });
};
/**
 * Adds the node_modules to the app module.
 * This is needed because we spawn different scripts when running start/build/test and so we lose
 * the original cwd directory
 */
exports.addCWDNodeModule = function () {
    var args = [];
    for (var _i = 0; _i < arguments.length; _i++) {
        args[_i] = arguments[_i];
    }
    var indexCoreCwd = args.indexOf('--core-cwd');
    if (indexCoreCwd !== -1) {
        var coreCwd = args[indexCoreCwd + 1];
        if (coreCwd) {
            exports.setCoreCwd(coreCwd);
        }
    }
    var indexCwd = args.indexOf('--cwd');
    if (indexCwd === -1) {
        // This is to setup the app environment
        exports.setCwd(exports.getCwd());
    }
    else {
        var cwd = args[indexCwd + 1];
        if (cwd) {
            exports.setCwd(cwd);
        }
    }
};
/**
 * Returns the absolute path to the pkg if found
 * @param pkg the package to lookup
 */
/* istanbul ignore next */
exports.resolveModulePath = function (pkg) {
    var paths = [];
    for (var _i = 1; _i < arguments.length; _i++) {
        paths[_i - 1] = arguments[_i];
    }
    try {
        return require.resolve(pkg);
    }
    catch (_a) {
        // Now try to specifically set the node_modules path
        var requirePaths = (require.main && require.main.paths) || [];
        requirePaths.push.apply(requirePaths, __spread(paths));
        try {
            return require.resolve(pkg, { paths: requirePaths });
        }
        catch (_b) {
            return false;
        }
    }
};
/**
 * Returns the path to flex-plugin-scripts
 */
exports._getFlexPluginScripts = function () {
    var flexPluginScriptPath = exports.resolveModulePath('flex-plugin-scripts');
    if (flexPluginScriptPath === false) {
        throw new Error('Could not resolve flex-plugin-scripts');
    }
    return path.join(path.dirname(flexPluginScriptPath), '..');
};
/**
 * Returns the path to flex-plugin-webpack
 */
exports._getFlexPluginWebpackPath = function (scriptsNodeModulesDir) {
    var flexPluginWebpackPath = exports.resolveModulePath('flex-plugin-webpack', scriptsNodeModulesDir);
    if (flexPluginWebpackPath === false) {
        throw new Error("Could not resolve flex-plugin-webpack");
    }
    return path.join(path.dirname(flexPluginWebpackPath), '..');
};
/**
 * Returns the paths to all modules and directories used in the plugin-builder
 */
// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
exports.getPaths = function () {
    var cwd = exports.getCwd();
    var nodeModulesDir = exports.resolveCwd('node_modules');
    var scriptsDir = exports._getFlexPluginScripts();
    var scriptsNodeModulesDir = exports.resolveRelative(scriptsDir, 'node_modules');
    var devAssetsDir = exports.resolveRelative(scriptsDir, 'dev_assets');
    var publicDir = exports.resolveCwd('public');
    var buildDir = exports.resolveCwd('build');
    var srcDir = exports.resolveCwd('src');
    var flexUIDir = exports.resolveRelative(nodeModulesDir, '@twilio/flex-ui');
    var tsConfigPath = exports.resolveCwd('tsconfig.json');
    var webpackDir = exports._getFlexPluginWebpackPath(scriptsNodeModulesDir);
    // package.json information
    var pkgName = '';
    var pkgVersion = '';
    // This file can be required in locations that don't have package.json
    try {
        var pkg = exports.readAppPackageJson();
        pkgName = pkg.name;
        pkgVersion = pkg.version;
    }
    catch (e) {
        // no-op
    }
    return {
        cwd: cwd,
        // flex-plugin-webpack paths
        webpack: {
            dir: webpackDir,
            nodeModulesDir: exports.resolveRelative(webpackDir, 'node_modules'),
        },
        // flex-plugin-scripts paths
        scripts: {
            dir: scriptsDir,
            nodeModulesDir: scriptsNodeModulesDir,
            devAssetsDir: devAssetsDir,
            indexHTMLPath: exports.resolveRelative(devAssetsDir, 'index.html'),
            tsConfigPath: exports.resolveRelative(devAssetsDir, 'tsconfig.json'),
        },
        // twilio-cli/flex/plugins.json paths
        cli: exports.getCliPaths(),
        // plugin-app (the customer app)
        app: {
            dir: cwd,
            name: pkgName,
            version: pkgVersion,
            pkgPath: exports.resolveCwd(packageJsonStr),
            jestConfigPath: exports.resolveCwd('jest.config.js'),
            webpackConfigPath: exports.resolveCwd('webpack.config.js'),
            devServerConfigPath: exports.resolveCwd('webpack.dev.js'),
            tsConfigPath: tsConfigPath,
            isTSProject: function () { return exports.checkFilesExist(tsConfigPath); },
            setupTestsPaths: [exports.resolveCwd('setupTests.js'), exports.resolveRelative(srcDir, 'setupTests.js')],
            // .env file support
            envPath: exports.resolveCwd('/.env'),
            hasEnvFile: function () { return exports.checkFilesExist(exports.resolveCwd('/.env')); },
            envExamplePath: exports.resolveCwd('/.env.example'),
            hasEnvExampleFile: function () { return exports.checkFilesExist(exports.resolveCwd('/.env.example')); },
            envDefaultsPath: exports.resolveCwd('/.env.defaults'),
            hasEnvDefaultsPath: function () { return exports.checkFilesExist(exports.resolveCwd('/.env.defaults')); },
            // build/*
            buildDir: buildDir,
            bundlePath: exports.resolveRelative(buildDir, pkgName, '.js'),
            sourceMapPath: exports.resolveRelative(buildDir, pkgName, '.js.map'),
            // src/*
            srcDir: srcDir,
            entryPath: exports.resolveRelative(srcDir, 'index'),
            // node_modules/*,
            nodeModulesDir: nodeModulesDir,
            flexUIDir: flexUIDir,
            flexUIPkgPath: exports.resolveRelative(flexUIDir, packageJsonStr),
            // public/*
            publicDir: publicDir,
            appConfig: exports.resolveRelative(publicDir, 'appConfig.js'),
            // dependencies
            dependencies: {
                react: {
                    version: exports.readPackageJson(exports.resolveRelative(nodeModulesDir, 'react', packageJsonStr)).version,
                },
                reactDom: {
                    version: exports.readPackageJson(exports.resolveRelative(nodeModulesDir, 'react-dom', packageJsonStr)).version,
                },
                flexUI: {
                    version: exports.readPackageJson(exports.resolveRelative(nodeModulesDir, '@twilio/flex-ui', packageJsonStr)).version,
                },
            },
        },
        // others
        assetBaseUrlTemplate: "/plugins/" + pkgName + "/%PLUGIN_VERSION%",
        extensions: ['js', 'mjs', 'jsx', 'ts', 'tsx'],
    };
};
/**
 * Returns the version of the dependency that is installed in node_modules
 * @param pkgName  the package name
 * @return the version of the package installed
 */
/* istanbul ignore next */
// eslint-disable-next-line import/no-unused-modules
exports.getDependencyVersion = function (pkgName) {
    try {
        return exports._require(pkgName + "/package.json").version;
    }
    catch (_a) {
        try {
            return exports._require(exports.resolveRelative(exports.getPaths().app.nodeModulesDir, pkgName, packageJsonStr)).version;
        }
        catch (_b) {
            return exports._require(exports.resolveRelative(exports.getPaths().scripts.nodeModulesDir, pkgName, packageJsonStr)).version;
        }
    }
};
/**
 * Returns the package.json version field of the package
 * @param name  the package
 */
/* istanbul ignore next */
exports.getPackageVersion = function (name) {
    var installedPath = exports.resolveRelative(exports.getPaths().app.nodeModulesDir, name, packageJsonStr);
    return exports.readPackageJson(installedPath).version;
};
