// flex-dev-utils/dist/validators.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.validateGitHubUrl = exports.isGitHubUrl = exports.isValidUrl = exports.validateApiKey = exports.validateAccountSid = exports.isInputNotEmpty = void 0;
var sids_1 = require("./sids");
var URL_REGEX = /^(https?:\/\/)?(www\.)?[a-z0-9]+([\-\.]{1}[a-z0-9]+)*\.[a-z]{2,5}(:[0-9]{1,5})?(\/.*)?$/;
var GITHUB_REGEX = /github\.com/;
/**
 * Simple input validator to verify it is not empty
 *
 * @param input the input to validate
 */
exports.isInputNotEmpty = function (input) { return Boolean(input && input.length > 0); };
/**
 * Validates that the accountSid is valid
 *
 * @param str the accountSid
 */
exports.validateAccountSid = function (str) {
    if (!exports.isInputNotEmpty(str)) {
        return false;
    }
    if (!sids_1.isSidOfType(str, sids_1.SidPrefix.AccountSid)) {
        return 'Invalid Account Sid was provided';
    }
    return true;
};
/**
 * Validates that the apiKey is valid
 *
 * @param str the apiKey
 */
exports.validateApiKey = function (str) {
    if (!exports.isInputNotEmpty(str)) {
        return false;
    }
    if (!sids_1.isSidOfType(str, sids_1.SidPrefix.ApiKey)) {
        return 'Invalid Account Sid was provided';
    }
    return true;
};
/**
 * Validates the string is valid URL
 *
 * @param url {string}  the URL string to validate
 * @return whether the URL is valid
 * @private
 */
exports.isValidUrl = function (url) { return URL_REGEX.test(url); };
/**
 * Validates the string is a GitHub URL
 *
 * @param url {string}  the URL string to validate
 * @return whether the URL is GitHub
 * @private
 */
exports.isGitHubUrl = function (url) { return GITHUB_REGEX.test(url); };
/**
 * Validates that the URL is a GitHub URL
 *
 * @param url the URL to validate
 */
exports.validateGitHubUrl = function (url) {
    if (!exports.isValidUrl(url)) {
        return 'Please enter a valid URL';
    }
    if (!exports.isGitHubUrl(url)) {
        return 'Only GitHub URLs are currently supported';
    }
    return true;
};
exports.default = {};
